unikernel.ml

```diff
@@ -490,39 +497,186 @@ struct
+                                store := store';
+                                let res =
+                                  "{\"status\": 200, \"success\": true, \
+                                   \"message\": {\"user\": "
+                                  ^ Yojson.Basic.to_string
+                                      (User_model.user_to_json user)
+                                  ^ "}}"
+                                in
+                                let cookie =
+                                  List.find
+                                    (fun (c : User_model.cookie) ->
+                                      c.name = "molly_session")
+                                    user.cookies
+                                in
+                                let cookie_value =
+                                  cookie.name ^ "=" ^ cookie.value
+                                  ^ ";Path=/;HttpOnly=true"
+                                in
+                                let header_list =
+                                  [
+                                    ("Set-Cookie", cookie_value);
+                                    ("location", "/dashboard");
+                                  ]
+                                in
+                                Lwt.return
+                                  (reply ~header_list
+                                     ~content_type:"application/json" res)
+                            | Error (`Msg _msg) ->
+                                let res =
+                                  "{\"status\": 400, \"success\": false, \
+                                   \"message\": \"Something went wrong. Wait a \
+                                   few seconds and try again.\"}"
+                                in
+                                Lwt.return
+                                  (reply ~content_type:"application/json" res)))
+                    ))
             | _ ->
                 let res =
                   "{\"status\": 400, \"success\": false, \"message\": \"Bad \
                    request method\"}"
                 in
                 Lwt.return (reply ~content_type:"application/json" res))
         | "/dashboard" ->
+            let _, (t : Storage.t) = !store in
+            let users = User_model.create_user_session_map t.users in
+            let middlewares = [ Middleware.auth_middleware ~users ] in
+            Middleware.apply_middleware middlewares
+              (fun _reqd ->
+                Lwt.return (reply ~content_type:"application/json" "Dashboard"))
+              reqd
+        | "/users" ->
+            let _, (t : Storage.t) = !store in
             Lwt.return
-              (reply ~content_type:"text/html"
-                 "This is going to be the Dashboard :-)")
+              (reply ~content_type:"application/json"
+                 (Yojson.Basic.to_string (Storage.t_to_json t)))
         | "/unikernel/create" ->
```

Comment:
here we as wel want to use the middleware to authenticate / ensure there's such a user

Reply:
oh this is just a helper endpoint for me to check when data changes. I must have committed on error :)